### ISSUE TITLE
stringImportPaths + $ENV_VAR always yield an error message, but works

### ISSUE THREAD

**Issue by p0nce** [dlang/dub#1504]
### System information

- **dub version**: DUB version 1.9.0, built on Jun  7 2018
- **OS Platform and distribution**: Windows 10
- **compiler version** DMD32 D Compiler v2.080.1

### Bug Description

It seems you can't use environment variable within stringImportPaths without having error messages.
Those error messages don't correspond to a real bug.

(We use this to convert C++ headers to D at compile-time.)

### How to reproduce?

Minimal `dub.json`

```json
{
    "name": "test-thing",
    "stringImportPaths": ["$MY_VAR"],
    "targetType": "none"
}
```
Set the `MY_VAR` environment variable to an existing, valid directory.

Output is:
```
C:\Users\ponce\Desktop\dub-test-case>dub
Invalid source/import path: C:\Users\ponce\Desktop\dub-test-case\$MY_VAR
Main package must have a binary target type, not none. Cannot build.
```

Same thing with `importPaths` and `sourcePaths`.

### Expected Behavior

I would expect the same output, but without the error message.

### Logs

C:\Users\ponce\Desktop\dub-test-case>dub -v
Using dub registry url 'https://code.dlang.org/'
Refreshing local packages (refresh existing: true)...
Looking for local package map at C:\ProgramData\dub\packages\local-packages.json
Looking for local package map at C:\Users\ponce\AppData\Local\dub\packages\local-packages.json
Note: Failed to determine version of package test-thing at .. Assuming ~master.
Refreshing local packages (refresh existing: false)...
Looking for local package map at C:\ProgramData\dub\packages\local-packages.json
Looking for local package map at C:\Users\ponce\AppData\Local\dub\packages\local-packages.json
Refreshing local packages (refresh existing: false)...
Looking for local package map at C:\ProgramData\dub\packages\local-packages.json
Looking for local package map at C:\Users\ponce\AppData\Local\dub\packages\local-packages.json
Checking for upgrades.
Using cached upgrade results...
Generating using build
Invalid source/import path: C:\Users\ponce\Desktop\dub-test-case\$MY_VAR
Configuring dependent test-thing, deps:
Main package must have a binary target type, not none. Cannot build.

**Comment by 0xEAB** [comment]
I can reproduce this "bug".

In case someone using Windows wants to quickly check it out himself, I wrote the following handy test script:
```batch
REM     Reproduction script for dlang/dub issue #1504

cd %~dp0

REM     Create stringImportDirectory
mkdir 1504
cd 1504
mkdir strImpDir

REM     Create example string file
echo Test string 1504... >strImpDir\xample1504.txt

REM     Create source files
mkdir source
echo import std.stdio;void main(){writeln(import("xample1504.txt"));} >%~dp0\1504\source\app.d

REM     Create DUB project
echo {"name":"test1504","stringImportPaths":["$DUB_ISSUE"]} >dub.json

REM     Set env var
set DUB_ISSUE=strImpDir

REM     Finally run the whole thing
dub -v
pause
```

**Comment by p0nce** [comment]
Fixed with #1569
